server/routerlicious/packages/protocol-base/src/quorum.ts

```diff
@@ -229,9 +226,57 @@ export class QuorumProposals extends TypedEventEmitter<IQuorumProposalsEvents> i
             throw new Error("Can't propose in disconnected state");
         }
 
-        const deferred = new Deferred<void>();
-        this.localProposals.set(clientSequenceNumber, deferred);
-        return deferred.promise;
+        return new Promise<void>((resolve, reject) => {
+            // The sequence number that our proposal went pending, which the msn must advance past.
+            // If undefined, then it's not sequenced yet.
+            let targetSequenceNumber: number | undefined;
+
+            // To resolve, a proposal goes through two phases:
+            // 1. Sequencing - waiting for the proposal to be ack'd, to learn the target msn value for approval.
+            // 2. Approval - waiting for the msn to advance past the target value.
+            const localProposalSequencedHandler = (sequencedCSN: number, sequenceNumber: number) => {
+                if (sequencedCSN === clientSequenceNumber) {
+                    targetSequenceNumber = sequenceNumber;
+                    this.stateEvents.off("localProposalSequenced", localProposalSequencedHandler);
+                    this.stateEvents.off("disconnected", disconnectedHandler);
```

Comment:
So, the proposal is still considered valid even if the client requesting the proposal disconnected before approval was completed ? I can see that theoretically we are just looking for msn after this, but curious if this is expected/valid use case scenario.

Reply:
Yes - generally speaking once a proposal gets sequenced, it no longer matters which client proposed it originally.  Even if every client disconnects before seeing the ack, when the next client joins they will see it and the proposal will get approved.